File: samples/dotnet-isolated-azurefunction/OuputBinding/SendMessageToKafka.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
// ------------------------------------------------------------

namespace dotnet_isolated_azurefunction
{
    using Microsoft.Azure.Functions.Worker;
    using Microsoft.Azure.Functions.Worker.Extensions.Dapr;
    using Microsoft.Extensions.Logging;
    using System.Text.Json;
    using System.Threading.Tasks;

    public static class SendMessageToKafka
    {
        [Function("SendMessageToKafka")]
        [DaprBindingOutput(BindingName = "%KafkaBindingName%", Operation = "create")]
        public static JsonElement Run(
            [DaprServiceInvocationTrigger] JsonElement payload, FunctionContext functionContext)
        {
            var log = functionContext.GetLogger("SendMessageToKafka");
            log.LogInformation("C#  function processed a SendMessageToKafka request.");

            return payload;
        }
    }
}

File: samples/java-azurefunction/src/main/java/com/function/SendMessageToKafka.java
package com.function;

import com.microsoft.azure.functions.ExecutionContext;
import com.microsoft.azure.functions.OutputBinding;
import com.microsoft.azure.functions.annotation.FunctionName;
import com.microsoft.azure.functions.dapr.annotation.DaprBindingOutput;
import com.microsoft.azure.functions.dapr.annotation.DaprServiceInvocationTrigger;

public class SendMessageToKafka {
    /**
     * This function gets invoked by dapr runtime:
     * dapr invoke --app-id functionapp --method SendMessageToKafka --data '{"data":{"message": "hello!" }}'
     * Once this function gets invoked, it will send a message to Kafka.
     */
    @FunctionName("SendMessageToKafka")
    public String run(
            @DaprServiceInvocationTrigger(
                name = "payload", 
                methodName = "SendMessageToKafka") 
            String payload,
            @DaprBindingOutput(
                name = "state",
                bindingName = "%KafkaBindingName%", 
                operation = "create")
            OutputBinding<String> product,
            final ExecutionContext context) {
        context.getLogger().info("Java  function processed a SendMessageToKafka request.");
        product.setValue(payload);

        return payload;
    }
}
